api4/channel.go

```diff
@@ -992,7 +992,7 @@ func getChannelByName(c *Context, w http.ResponseWriter, r *http.Request) {
 		}
 	} else {
 		if !c.App.SessionHasPermissionToChannel(c.App.Session, channel.Id, model.PERMISSION_READ_CHANNEL) {
-			c.SetPermissionError(model.PERMISSION_READ_CHANNEL)
+			c.Err = model.NewAppError("getChannelByName", store.MISSING_CHANNEL_ERROR, nil, "teamId="+channel.TeamId+", "+"name="+channel.Name+"", http.StatusNotFound)
```

Comment:
Is better this way to concatenate strings or the usage of fmt.Sprintf?

Reply:
I guess since all the data to print is of type `string` this way you avoid calling an extra function. I agree it's a bit ugly but it appears to be the accepted way since it's all over the codebase. Maybe @jespino knows more about this.

Reply:
Is nothing super important so you can merge if this needs to be merged, is more a question about how are we doing this kind of interpolation. Reading about it the `+` is the fastest so go ahead it's good for me